secure-ios-app/storage/StorageViewController.swift

```diff
@@ -192,9 +197,15 @@ class StorageViewController: UITableViewController {
      - Function to load/re-load notes to display in the UI
      */
     func loadNotes() {
-        let storedNotes = self.storageListener?.list()
-        self.notes = storedNotes!
-        self.tableView.reloadData()
+        DispatchQueue.global(qos: .background).sync {
+            DispatchQueue(label: "background").sync {
+                let storedNotes = self.storageListener?.list()
+                self.notes = storedNotes!
+            }
+            DispatchQueue(label: "ui").sync {
+                self.tableView.reloadData()
```

Comment:
I see here another 2 instances of `DispatchQueue` are created and each of them is processing some task. In this case, I think it's not guaranteed that the notes will be the latest when the `tableView` is reloaded. Not sure if that is intended?

Reply:
Also I don't think the tasks submitted to the `ui` queue are not actually executed on the main UI thread, it's a bit confusing.